native/src/routes/ChangeLanguageModal.tsx

```diff
@@ -13,56 +11,46 @@ import SelectorItemModel from '../models/SelectorItemModel'
 const Wrapper = styled.ScrollView`
   background-color: ${props => props.theme.colors.backgroundColor};
 `
+
 type PropsType = {
-  theme: ThemeType
-  currentLanguage: string
-  languages: Array<LanguageModel>
-  availableLanguages: Array<string>
-  changeLanguage: (newLanguage: string) => void
   route: RoutePropType<ChangeLanguageModalRouteType>
   navigation: NavigationPropType<ChangeLanguageModalRouteType>
 }
 
-class ChangeLanguageModal extends React.Component<PropsType> {
-  onPress = (model: LanguageModel): void => {
-    const { changeLanguage } = this.props
-    this.closeModal()
-    InteractionManager.runAfterInteractions(() => {
-      changeLanguage(model.code)
+const ChangeLanguageModal = ({ navigation, route }: PropsType): ReactElement => {
+  const { currentLanguage, languages, availableLanguages, cityCode } = route.params
+  const dispatch = useDispatch()
+
+  const changeLanguage = (newLanguage: string) => {
+    dispatch({
+      type: 'SWITCH_CONTENT_LANGUAGE',
+      params: {
+        newLanguage,
+        city: cityCode
+      }
     })
   }
 
-  closeModal = (): void => {
-    const { navigation } = this.props
-    navigation.goBack()
-  }
+  const selectorItems = languages.map(({ code, name }) => {
+    const isLanguageAvailable = availableLanguages.includes(code)
+    return new SelectorItemModel({
+      code,
+      name,
+      enabled: isLanguageAvailable,
+      onPress: () => {
+        if (code !== currentLanguage) {
+          changeLanguage(code)
+        }
```

Comment:
This is new. Also removed `InteractionManager` as the following is definitely not true for a simple dispatch:
> InteractionManager allows long-running work to be scheduled after any interactions/animations have completed. In particular, this allows JavaScript animations to run smoothly.